shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -61,8 +61,13 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
 // Returns an empty array if no spell check suggestions.
 - (NSArray<NSDictionary<NSString*, id>*>*)findAllSpellCheckSuggestionsForText:(NSString*)text
                                                                    inLanguage:(NSString*)language {
-  // Transform Dart Locale format to iOS language format.
-  language = [language stringByReplacingOccurrencesOfString:@"-" withString:@"_"];
+  // Transform Dart Locale format to iOS language format if necessary.
+  if ([language containsString:@"-"]) {
+    NSArray<NSString*>* languageCodes = [language componentsSeparatedByString:@"-"];
+    NSString* lastCode = [[languageCodes lastObject] uppercaseString];
```

Comment:
nits: 
Add `FML_DCHECK(languageCodes.count == 2);` to catch error when dart format style changes.